src/storage/src/hummock/local_version_manager.rs

```diff
@@ -395,6 +395,13 @@ impl LocalVersionManager {
         is_remote_batch: bool,
         table_id: u32,
     ) -> HummockResult<usize> {
+        let max_sync_epoch = self.local_version.read().get_max_sync_epoch();
+        assert!(
+            epoch > max_sync_epoch,
+            "write epoch must greater than sync epoch, write epoch{}, sync epoch{}",
+            epoch,
+            max_sync_epoch
+        );
```

Comment:
We can do the check in `write_shared_buffer_inner` to avoid acquiring the read lock.